staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go

```diff
@@ -456,6 +460,73 @@ func (c *$.type|privatePlural$) List(opts $.ListOptions|raw$) (result *$.resultT
 		timeout = time.Duration(*opts.TimeoutSeconds) * time.Second
 	}
 	result = &$.resultType|raw$List{}
+
+	wgLen := 1
+	// When resource version is not empty, it reads from api server local cache
+	// Need to check all api server partitions
+	if opts.ResourceVersion != "" && len(c.clients) > 1 {
+		wgLen = len(c.clients)
+	}
+
+	if wgLen > 1 {
+		var listLock sync.Mutex
+
+		var wg sync.WaitGroup
+		wg.Add(wgLen)
+		results := make(map[int]*$.resultType|raw$List)
+		errs := make(map[int]error)
+		for i, client := range c.clients {
+			go func(c *$.type|privatePlural$, ci $.RESTClientInterface|raw$, opts $.ListOptions|raw$, lock $.syncMutex|raw$, pos int, resultMap map[int]*$.resultType|raw$List, errMap map[int]error) {
+				r := &$.resultType|raw$List{}
+				err := ci.Get().
+					$if .namespaced$Tenant(c.te).Namespace(c.ns).$end$
+					$if .tenanted$Tenant(c.te).$end$
+					Resource("$.type|resource$").
+					VersionedParams(&opts, $.schemeParameterCodec|raw$).
+					Timeout(timeout).
+					Do().
+					Into(r)
+
+				lock.Lock()
+				resultMap[pos] = r
+				errMap[pos] = err
+				lock.Unlock()
+				wg.Done()
+			}(c, client, opts, listLock, i, results, errs)
+		}
+		wg.Wait()
+
+		// consolidate list result
+		itemsMap := make(map[string]*$.resultType|raw$)
+		for j:=0; j < wgLen; j++ {
+			currentErr, isOK := errs[j]
+			if isOK && currentErr != nil {
+				if !($.errorIsForbidden|raw$(currentErr) && $.stringsContains|raw$(currentErr.Error(), "no relationship found between node")) {
+					err = currentErr
+					return
+				} else {
+					continue
+				}
+			}
+
+			currentResult, _ := results[j]
+			if result.Kind == "" {
+				result.TypeMeta = currentResult.TypeMeta
+				result.ListMeta = currentResult.ListMeta
```

Comment:
I guess you add the check "if result.kind == """ if for efficiency concern. If so, you need set the result.Kind in the block after that check.